src/unix/mediactrl_gstplayer.cpp

```diff
@@ -451,12 +501,20 @@ wxLongLong wxGStreamerMediaBackend::GetDuration()
     return GST_CLOCK_TIME_IS_VALID(duration) ? duration / GST_MSECOND : 0;
 }
 
-void wxGStreamerMediaBackend::Move(int WXUNUSED(x), int WXUNUSED(y), int WXUNUSED(w), int WXUNUSED(h))
+void wxGStreamerMediaBackend::Move(int WXUNUSED(x), int WXUNUSED(y), int w, int h)
 {
-    /* Nothing to be done here, at least for GTK+. For other toolkits we might
-     * have to call
-     * gst_player_video_overlay_video_renderer_set_render_rectangle() here
+    /* GTK+ on X11 does not require explicitely setting the video size.
+     * Other toolkits might need to do something similar to GTK wayland too.
      */
+#ifdef __WXGTK3__
+    GdkWindow* window = gtk_widget_get_window(m_ctrl->m_wxwindow);
+    if (window && strcmp("GdkWaylandWindow", g_type_name(G_TYPE_FROM_INSTANCE(window))) == 0) {
+        int x, y;
+        gdk_window_get_origin(window, &x, &y);
+        gst_player_video_overlay_video_renderer_set_render_rectangle(GST_PLAYER_VIDEO_OVERLAY_VIDEO_RENDERER(m_video_renderer),
+                                                                         x, y, w, h);
```

Comment:
Another wraparound suggestion:
```suggestion
        gst_player_video_overlay_video_renderer_set_render_rectangle(
            GST_PLAYER_VIDEO_OVERLAY_VIDEO_RENDERER(m_video_renderer),
            x, y, w, h
        );
```